modules/javafx.controls/src/main/java/javafx/scene/control/Skinnable.java

```diff
@@ -42,9 +42,8 @@ public interface Skinnable {
      * <p>
      * Some implementations of {@code Skinnable} define a one-to-one relationship between {@code Skinnable}
      * and its {@code Skin}. Every {@code Skin} maintains a back reference to the
-     * {@code Skinnable}.  When required, this relationship is enforced inside {@link Skinnable#setSkin(Skin)} method
-     * which checks the return value of {@link Skin#getSkinnable()} against this {@code Skinnable},
-     * throwing an {@code IllegalArgumentException} if it is not the same.
+     * {@code Skinnable}.  When required, this relationship is enforced when the {@code Skin} is set,
+     * throwing an {@code IllegalArgumentException}.
```

Comment:
It looks like you removed too much. I think you need to retain the information about what will cause the exception. Maybe something like `,throwing an {@code IllegalArgumentException} if the return value of {@link Skin#getSkinnable()} is not the same as this {@code Skinnable}.`?

Reply:
corrected, thanks!

Reply:
I sort of left that out on purpose, since I think that's too much detail for an interface description.  The relationship is enforced, and it's not up to the docs to describe the exact mechanism (or future mechanisms).